File: backend/app/models/assignment.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Boolean, ForeignKey, UniqueConstraint, func
from app.database import Base

class Assignment(Base):
    __tablename__ = "assignments"
    id = Column(Integer, primary_key=True, index=True)
    course_id = Column(Integer, ForeignKey("courses.id"), nullable=False)
    title = Column(String(200), nullable=False)
    description = Column(Text)
    due_date = Column(DateTime(timezone=True))
    max_score = Column(Integer, server_default="100")
    is_active = Column(Boolean, server_default="true")
    file_url = Column(String(500))
    file_name = Column(String(200))
    created_at = Column(DateTime(timezone=True), server_default=func.now())

class Submission(Base):
    __tablename__ = "submissions"
    __table_args__ = (UniqueConstraint("assignment_id", "user_id"),)
    id = Column(Integer, primary_key=True, index=True)
    assignment_id = Column(Integer, ForeignKey("assignments.id"), nullable=False)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    file_url = Column(String(500))
    comments = Column(Text)
    score = Column(Integer)
    graded = Column(Boolean, server_default="false")
    feedback = Column(Text)
    submitted_at = Column(DateTime(timezone=True), server_default=func.now())
    graded_at = Column(DateTime(timezone=True))

File: frontend-next/types/assignment.ts
import type { components } from '@/lib/api/schema';

export type Assignment = components['schemas']['AssignmentOut'];

export type AssignmentSubmission = {
  id: number;
  assignment_id: number;
  user_id: number;
  student_name?: string | null;
  student_email?: string | null;
  file_url?: string | null;
  file_name?: string | null;
  comments?: string | null;
  grade?: number | null;
  feedback?: string | null;
  submitted_at?: string | null;
  graded_at?: string | null;
};
